tests/e2e/runner/__tests__/minter-suite-v2/ram-lib.test.ts

```diff
@@ -0,0 +1,750 @@
+
+      await minterRAMV0Contract.connect(artist).setAuctionDetails(
+        0, // _projectId
+        genArt721CoreAddress, // _coreContract
+        targetAuctionStart, // _auctionTimestampStart
+        targetAuctionEnd, // _auctionTimestampEnd
+        ethers.utils.parseEther("1"), // _basePrice
+        true, // _allowExtraTime
+        true // _adminArtistOnlyMintPeriodIfSellout
+      );
+      // validate project minter config in subgraph
+      await waitUntilSubgraphIsSynced(client);
+      // Manually limit project max invocations
+      // @dev this can only happen in RAM Auction State A
+      await minterRAMV0Contract
+        .connect(artist)
+        .manuallyLimitProjectMaxInvocations(
+          0, // _projectId
+          genArt721CoreAddress, // _coreContract
+          1000
+        );
+
+      await waitUntilSubgraphIsSynced(client);
+      const targetId = `${minterRAMV0Address.toLowerCase()}-${genArt721CoreAddress.toLowerCase()}-0`;
+      const minterConfigRes = await getProjectMinterConfigurationDetails(
+        client,
+        targetId
+      );
+      // validate extraMinterDetails
+      const extraMinterDetails = JSON.parse(minterConfigRes.extraMinterDetails);
+      expect(extraMinterDetails.auctionNumTokensInAuction).toBe("1000");
+    });
+  });
+
+  describe("BidCreated, BidToppedUp", () => {
+    afterEach(async () => {
+      try {
+        await sharedMinterFilterContract
+          .connect(artist)
+          .removeMinterForProject(0, genArt721CoreAddress);
+      } catch (error) {
+        // swallow error in case of test failure
+      }
+    });
+
+    test("subgraph is updated after event emitted", async () => {
+      // artist configures the auction
+      // @dev set minter as active
+      await sharedMinterFilterContract.connect(artist).setMinterForProject(
+        0, // _projectId
+        genArt721CoreAddress, // _coreContract
+        minterRAMV0Address // _minter
+      );
+
+      const latestBlock = await deployer.provider.getBlock("latest");
+      const targetAuctionStart = latestBlock.timestamp + 60;
+      const targetAuctionEnd = targetAuctionStart + 600;
+      // Set auction details
+      await minterRAMV0Contract.connect(artist).setAuctionDetails(
+        0, // _projectId
+        genArt721CoreAddress, // _coreContract
+        targetAuctionStart, // _auctionTimestampStart
+        targetAuctionEnd, // _auctionTimestampEnd
+        ethers.utils.parseEther("1"), // _basePrice
+        true, // _allowExtraTime
+        true // _adminArtistOnlyMintPeriodIfSellout
+      );
+      // validate project minter config in subgraph
+      await waitUntilSubgraphIsSynced(client);
+      // Get slot index for bid value
+      const slot10price = await minterRAMV0Contract.slotIndexToBidValue(
+        0, //_projectId
+        genArt721CoreAddress, // _coreContract
+        10 // _slotIndex
+      );
+      // Wait until auction starts (60s + 1s margin)
+      await new Promise((resolve) => {
```

Comment:
I think it might be possible to use some hardhat network specific json rpc methods to jump ahead and avoid the wait here. Not totally sure how that will work with the graph node but probably worth a try. see https://hardhat.org/hardhat-network/docs/reference#evm_setnextblocktimestamp

Reply:
Thanks for flagging! Seems to work using `evm_increaseTime` - I added a helper in our test helpers file.